controller/control.go

```diff
@@ -160,17 +161,53 @@ func (c *Controller) hasWOReplica() (string, bool) {
 	return "", false
 }
 
+func (c *Controller) hasGreaterRevisionCount(woReplica, newReplica string) (bool, error) {
+	woRevCnt, err := c.backend.GetRevisionCounter(woReplica)
+	if err != nil {
+		return false, err
+	}
+
+	repClient, err := replicaClient.NewReplicaClient(newReplica)
+	if err != nil {
+		return false, err
+	}
+
+	replica, err := repClient.GetReplica()
+	if err != nil {
+		return false, err
+	}
+
+	newRevCnt, err := strconv.ParseInt(replica.RevisionCounter, 10, 64)
+	if err != nil {
+		return false, err
+	}
+	if newRevCnt > woRevCnt {
+		return true, nil
+	}
+	return false, nil
+}
```

Comment:
Let's avoid rest calls with lock.

Reply:
I am avoiding the rest call if revision count is already available in the registered replica for woReplica, for newReplica it maynot be in RegisteredReplicas